build-farm/platform-specific-configurations/downloaders.sh

```diff
@@ -0,0 +1,147 @@
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+####################################################################################
+# This file is gathering all download boot jdk functions, so they can b reused later
+# On long run, the methods - due theirs simialrity - should converge to one
+####################################################################################
+
+# this one should be swithing per (supported) oses in future
+function downloadBootJDK() {
+  if uname -o | grep -i -e Linux ; then
+    downloadLinuxBootJDK "$@"
+  elif uname -o | grep -i -e Cygwin -e Windows ; then
+    downloadWindowsBootJDK "$@"
+  else
+    echo "Unsupported platfrom for direct download of boot jdk: `uname -m` `uname -o`"
+    exit 1
+  fi
+}
+
+function downloadLinuxBootJDK() {
+  ARCH=$1
+  VER=$2
+  export downloadArch
+  case "$ARCH" in
+     "riscv64") downloadArch="$NATIVE_API_ARCH";;
+             *) downloadArch="$ARCH";;
+  esac
+  releaseType="ga"
+  vendor="eclipse"
+  apiUrlTemplate="https://api.adoptium.net/v3/binary/latest/\${VER}/\${releaseType}/linux/\${downloadArch}/jdk/hotspot/normal/\${vendor}"
+  apiURL=$(eval echo "${apiUrlTemplate}")
+  echo "Downloading GA release of boot JDK version ${VER} from ${apiURL}"
+  # make-adopt-build-farm.sh has 'set -e'. We need to disable that for
+  # the fallback mechanism, as downloading of the GA binary might fail.
+  set +e
+  curl -L -o bootjdk.tar.gz "${apiURL}"
+  apiSigURL=$(curl -v "${apiURL}" 2>&1 | tr -d \\r | awk '/^< Location:/{print $3 ".sig"}')
+  if ! grep "No releases match the request" bootjdk.tar.gz; then
+    curl -L -o bootjdk.tar.gz.sig "${apiSigURL}"
+    gpg --keyserver keyserver.ubuntu.com --recv-keys 3B04D753C9050D9A5D343F39843C48A565F8F04B
+    echo -e "5\ny\n" |  gpg --batch --command-fd 0 --expert --edit-key 3B04D753C9050D9A5D343F39843C48A565F8F04B trust;
+    gpg --verify bootjdk.tar.gz.sig bootjdk.tar.gz || exit 1
+    mkdir "$bootDir"
+    tar xpzf bootjdk.tar.gz --strip-components=1 -C "$bootDir"
+    set -e
+  else
+    # We must be a JDK HEAD build for which no boot JDK exists other than
+    # nightlies?
+    echo "Downloading GA release of boot JDK version ${VER} failed."
+    # shellcheck disable=SC2034
+    releaseType="ea"
+    # shellcheck disable=SC2034
+    vendor="adoptium"
+    apiURL=$(eval echo ${apiUrlTemplate})
+    echo "Attempting to download EA release of boot JDK version ${VER} from ${apiURL}"
+    set +e
+    curl -L -o bootjdk.tar.gz "${apiURL}"
+    if ! grep "No releases match the request" bootjdk.tar.gz; then
```

Comment:
seems like duplicated logic from above, could extract?

Reply:
It is utterly duplicated. Most of body of the two extracted methods is very similar. I had really only re-factored linux.sh and windows.sh  boot jdk download to downloaders.sh.

It would be worthy to revisit those methods as separate PR, to unify as much logic as necessary.  AFAICJ all download methods should be processed like this, before the single replacement is crated.

If you wish, I can do some minor fixes (As the one you suggest here) but that may be contra productive in future.